### DISCUSSION TITLE
[Feature Request]: Email address obfuscation/encryption

### DISCUSSION THREAD

**Discussion by brzGatsu** [GetPublii/Publii#1992] [Ideas]
### Feature Description

It would be great to have a builtin email-address obfuscation/encryption as spam-prevention.

Here are some common techniques:
https://www.ionos.com/digitalguide/e-mail/e-mail-security/protecting-your-email-address-how-to-do-it/

I personally like the javascript/rot13 solution but with a random rotation, not 13. It can be combined with hex entity conversion as well.

Captcha would in most cases require an external provider which is problematic in terms of gdpr.

**Comment by candidexmedia** [comment]
@brzGatsu Thanks for sharing the link! I didn't know what rot13 was until now.

In the meantime, I use a form with a honeypot. It doesn't eliminate all spam, but at least the address (whether scrambled or not) isn't out there.

I've also heard of Mailgo, although I'm not sure if their "less spam" implementation is truly scraper safe...

https://mailgo.dev/docs/examples

**Comment by brzGatsu** [comment]
Great to see that there is a official email obfuscation plugin now. Two things regarding that:
1. I am not sure how effective it is, since it automatically unobfuscates when the DOM is loaded. This should be easy to circumvent for crawlers, no? The way I did it: By default it says something like "Contact: click to see Email". This link has a click handler that calls a javascript which then performs the unobfuscation, opens the mail program and also replaces the the "click to see" text with the actual address.
2. You could add an additional layer of obfuscation to all three methods, which the plugin offers, by replacing the characters by hex entities.

**Comment by raramuridesign** [comment]
@brzGatsu We are using this plugin and have no issues - I think the 3 different options are perfect and offer a solution for everyone - 
https://marketplace.getpublii.com/plugins/email-address-obfuscation/ 

@dziudek just for interest ---
